File: patterns/linked_lists/basic_operations/list_size.py
from patterns.linked_lists.lib import Node


# Time: O(n)
# Memory: O(1)
def size(head: Node) -> int:
    count: int = 0
    curr: Node = head

    while curr is not None:
        count += 1
        curr = curr.next

    return count


if __name__ == "__main__":
    linked_list: Node = Node(1, Node(2, Node(3, None)))
    print(f"Size: {size(linked_list)}")

File: patterns/linked_lists/basic_operations/list_size.cpp
#include "./../lib.h"
#include <iostream>

using namespace std;

int size(Node<int> *head)
{
    int count = 0;
    Node<int> *curr = head;

    while (curr != nullptr)
    {
        count++;
        curr = curr->next;
    }

    return count;
}

int main()
{
    Node<int> *head = new Node<int>(1, new Node<int>(2, new Node<int>(3, nullptr)));

    cout << "Size: " << size(head) << endl;

    delete head;
    return 0;
}